test/run/test_mount.py

```diff
@@ -67,56 +65,106 @@ def test_mount_ro(tmpdir):
+def test_without_options(tmpdir):
+    tree, size = create_image(tmpdir)
+    options = {}
+
+    with tempfile.TemporaryDirectory(dir=tmpdir) as mountpoint:
         with host.ServiceManager() as mgr:
             with make_dev_tmpfs(tmpdir) as devpath:
-                # Device manager to open the loopback device
-                devmgr = devices.DeviceManager(mgr, devpath, tree)
-                # get a Device for the loopback
-                dev = devices.Device(
-                    "loop",
-                    index.get_module_info(
-                        "Device",
-                        "org.osbuild.loopback"
-                    ),
-                    None,
-                    {
-                        "filename": "image.img",
-                        "start": 0,
-                        "size": size // 512  # size is in sectors / blocks
-                    }
-                )
-                # open the device and get its loopback path
-                lpath = os.path.join(
-                    devpath,
-                    devmgr.open(dev)["path"]
-                )
-                # mount the loopback
-                mounts.MountManager(
-                    devmgr,
-                    mountpoint
-                ).mount(
-                    mounts.Mount(
-                        lpath,
-                        index.get_module_info("Mount", "org.osbuild.ext4"),
-                        dev,
-                        "/",
-                        {"readonly": True}
-                    )
-                )
-
-                # try to write in it, and failure to do so is a success
+                mount(mgr, devpath, tree, size, mountpoint, options)
+                with open(os.path.join(mountpoint, "test"), "w", encoding="utf-8") as f:
+                    f.write("should work")
+                os.remove(os.path.join(mountpoint, "test"))
+
+
+@pytest.mark.skipif(not TestBase.can_bind_mount(), reason="root only")
+def test_all_options(tmpdir):
+    tree, size = create_image(tmpdir)
+    options = {
+        "readonly": True,
+        "uid": 0,
+        "gid": 0,
+        "umask": "077",
+        "shortname": "winnt"
+    }
+    print(options)
+
+    with tempfile.TemporaryDirectory(dir=tmpdir) as mountpoint:
+        with host.ServiceManager() as mgr:
+            with make_dev_tmpfs(tmpdir) as devpath:
+                mount(mgr, devpath, tree, size, mountpoint, options)
+
+                # Check FS is read only
                 with pytest.raises(OSError) as err:
                     with open(os.path.join(mountpoint, "test"), "w", encoding="utf-8") as f:
                         f.write("should not work")
-                assert err.value.errno == errno.EROFS  # Check that FS is read only
+                assert err.value.errno == errno.EROFS
+
+                # Check the other options
+                st = os.lstat(mountpoint)
+                assert filemode(st.st_mode) == "drwx------"
```

Comment:
Could just compare to the octal representation `0o40700`.  It would still be mostly readable and wouldn't need the extra import.  It's test code though so I don't mind.

Reply:
That'll look better 👍